turbopack/crates/turbopack-ecmascript/src/references/esm/export.rs

```diff
@@ -538,9 +539,11 @@ impl EsmExports {
 impl EsmExports {
     pub async fn code_generation(
         self: Vc<Self>,
-        _module_graph: Vc<ModuleGraph>,
+        module_graph: Vc<ModuleGraph>,
         chunking_context: Vc<Box<dyn ChunkingContext>>,
+        module: Vc<Box<dyn EcmascriptChunkPlaceable>>,
         parsed: Option<Vc<ParseResult>>,
+        remove_unused_exports: bool,
     ) -> Result<CodeGeneration> {
         let expanded = self.expand_exports().await?;
```

Comment:
expand_exports can benefit from "used exports info". It only need to expand star exports if the used exports are not yet handled by the direct exports. Dynamic reexports only need to be generated when static star exports don't cover used exports